### ISSUE TITLE
Diffing is slow

### ISSUE THREAD

**Issue by ding-an-sich** [rrrene/credo#949]
### Environment

* Credo version (`mix credo -v`): 1.6.1
* Erlang/Elixir version (`elixir -v`): Elixir 1.13.1 (compiled with Erlang/OTP 24)
* Operating system: Linux x86_64

### What were you trying to do?

Using the diff command to only run credo checks on changed files.

### Expected outcome

Noticeable reduction in running time when checking only for changed files.

### Actual outcome

As can be seen below `mix credo diff` and `mix credo` take a similar time within a large codebase.

Manually passing the output of `git diff` to the `mix credo --files-included`command produces a very noticeable decrease in running time.

```bash
➜ MIX_ENV=test mix credo diff --from-git-merge-base origin/master --strict
Diffing 1219 source files in working dir with origin/master (this might take a while) ...
Analysis took 72.4 seconds (1.3s to load, 71.0s running 67 checks on 1219 files)

Changes between 1d09cd7e7c84396c0c45866c8b5e78df62563f79 and working dir:

+  added no issues,
✔  fixed 4 refactoring opportunities, 37 software design suggestions, and
~  kept 4 refactoring opportunities, 37 software design suggestions.

------------------------------------------------------------------------------------------------

➜ MIX_ENV=test mix credo --strict
Checking 1219 source files (this might take a while) ...
Analysis took 70.2 seconds (2.1s to load, 68.1s running 67 checks on 1219 files)
7427 mods/funs, found 4 refactoring opportunities, 37 software design suggestions.

------------------------------------------------------------------------------------------------

➜ MIX_ENV=test CHANGES=`git diff master --name-only | grep '.exs\|.ex' | awk -v d=" --files-included " '{s=(s d)$0}END{print s}'`; [ ! -z "$CHANGES" ] &&  mix credo --strict `echo $CHANGES`
Checking 2 source files ...
Analysis took 0.2 seconds (0.1s to load, 0.1s running 67 checks on 2 files)
33 mods/funs, found no issues.
```

**Comment by rrrene** [comment]
This is by design. Credo's Diff command does not run Credo on the changed files (as you expect), but it diffs the results of two revisions (the given one and HEAD). 

The reason for this is simple: If we just look at changed files from Git, we get issues for functions we did not touch (but we changed something else in that file). 

Example:

You have a file with ten functions and five Credo issues. You commit. Then you change a comment and run your version of Credo diff, which produces the five issues from before

That was not desirable.
